src/Transport/Topology/NumberOfTopicsInBundleCheck.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace NServiceBus.Transport.AzureServiceBus
+{
+    using System.Linq;
+    using System.Threading.Tasks;
+    using NServiceBus.AzureServiceBus.Topology.MetaModel;
+
+    static class NumberOfTopicsInBundleCheck
+    {
+        public static async Task<NamespaceBundleConfigurations> Run(IManageNamespaceManagerLifeCycleInternal manageNamespaceManagerLifeCycle, NamespaceConfigurations namespaceConfigurations, string bundlePrefix)
+        {
+            var namespaceBundleConfigurations = new NamespaceBundleConfigurations();
+
+            foreach (var namespaceConfiguration in namespaceConfigurations)
+            {
+                var namespaceManager = manageNamespaceManagerLifeCycle.Get(namespaceConfiguration.Alias);
+                var namespaceManagerThatCanQueryAndFilterTopics = namespaceManager as NamespaceManagerAdapterInternal;
```

Comment:
This is exactly why I had that TODO comment [here](https://github.com/Particular/NServiceBus.AzureServiceBus/pull/570#discussion_r120271310) 😃